src/hist/intervals.py

```diff
@@ -55,11 +55,12 @@ def poisson_interval(
     # https://github.com/CoffeaTeam/coffea/blob/8c58807e199a7694bf15e3803dbaf706d34bbfa0/LICENSE
     if coverage is None:
         coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
-    scale = np.empty_like(values)
-    scale[values != 0] = variances[values != 0] / values[values != 0]
+    scale = np.ones_like(values)
+    mask = np.isfinite(values) & (values != 0)
+    scale[mask] = variances[mask] / values[mask]
```

Comment:
```suggestion
    np.divide(variances, values, out=scale, where=mask)
```

This will be _much_ faster. Mask selection is slow.